src/lib/protocols/munin.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * munin.c
+ *
+ * Copyright (C) 2022 - ntop.org
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_MUNIN
+
+#include "ndpi_api.h"
+
+static void ndpi_int_munin_add_connection(struct ndpi_detection_module_struct * const ndpi_struct,
+                                          struct ndpi_flow_struct * const flow)
+{
+  NDPI_LOG_INFO(ndpi_struct, "found munin\n");
+
+  ndpi_set_detected_protocol(ndpi_struct, flow,
+                             NDPI_PROTOCOL_MUNIN,
+                             NDPI_PROTOCOL_UNKNOWN,
+                             NDPI_CONFIDENCE_DPI);
+}
+
+/* ***************************************************** */
+
+void ndpi_search_munin(struct ndpi_detection_module_struct *ndpi_struct,
+                       struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+  static char const munin_prefix[] = "# munin node at ";
+
+  NDPI_LOG_DBG(ndpi_struct, "search munin\n");
+
+  // "# munin node at "
+  if (packet->payload_packet_len < NDPI_STATICSTRING_LEN(munin_prefix))
+  {
+    NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    return;
+  }
+
+  if (memcmp(packet->payload, munin_prefix, NDPI_STATICSTRING_LEN(munin_prefix)) != 0)
+  {
+    NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    return;
+  }
+
+  ndpi_int_munin_add_connection(ndpi_struct, flow);
+  flow->host_server_name[0] = '\0';
+
+  if (packet->payload[packet->payload_packet_len - 1] != '\n')
+  {
+    ndpi_set_risk(ndpi_struct, flow, NDPI_MALFORMED_PACKET, "Missing Munin Hostname");
+    return;
+  }
+
+  size_t host_len = packet->payload_packet_len - NDPI_STATICSTRING_LEN(munin_prefix) - 1;
+  if (host_len > 0)
```

Comment:
It seems wrong... if packet->payload_packet_len == NDPI_STATICSTRING_LEN(munin_prefix), the `if` condition is true...
Having `host_len` as a **signed** integer?

Reply:
Under those circumstances it should already bail out at `packet->payload[packet->payload_packet_len - 1] != '\n'` or am I wrong?

Reply:
no, you are right. Thanks